jsf/subsystem/src/test/java/org/jboss/as/jsf/deployment/JSFDepProcessorTestCase.java

```diff
@@ -0,0 +1,57 @@
+package org.jboss.as.jsf.deployment;
+
+import org.jboss.as.server.deployment.AttachmentKey;
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
+import org.jboss.as.server.deployment.module.ModuleSpecification;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@PowerMockIgnore("jdk.internal.reflect.*")
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({JsfVersionMarker.class, JSFModuleIdFactory.class})
+public class JSFDepProcessorTestCase {
```

Comment:
I think it would be better to create a deployment that's going to fail there. The deployment will fail if JSF has been disabled for it. The method that's used for determining whether or not JSF should be activated for a deployment can be found here:

https://github.com/wildfly/wildfly/blob/master/jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFVersionProcessor.java#L149-L164

From that method, we can see that if the deployment isn't a WAR then JSF should not be activated. So it looks like it should be possible to create a test that sets the `default-jsf-impl-slot` to some invalid value and then attempts to deploy an EAR.

Reply:
I think this can be tested in Arquillian with something like: https://github.com/spyrkob/wildfly/commit/9acf4ad3df71d5fcadebb1bafefacd271ef2c2c9

Reply:
@bstansberry I have tested Arquilian test suggested by @spyrkob, its working. The test case which I have written using Mockito is actually testing the code blocks of JSFDependencyProcessor.java . Please suggest if you are okay with the arquilian test then I'll move the changes.

Reply:
@moulalis Sorry; I've let this one sit idle. I prefer an arquillian integration test vs complex use of mocks. Thanks.

Reply:
@bstansberry I've added the changes with arquillian.